### PR TITLE
added convergence criteria

### PR DESCRIPTION
**Pull request by SampsonML**
Here, we can directly access the model updates from the optimiser which is equivalent to (x_old - x). This bypasses the need to look at gradients directly, which was what I was doing previously. If the L2 norm of the updates is less than the user-defined relative tolerance of the current data then we have converged.

### CODE REVIEW

File: scarlet2/scene.py [hunk 1]
```diff
@@ -175,23 +175,32 @@ def update_fn(updates, state, params):
...
-
-                # terminate optimization if all parameter change less than e_rel
+                
+                # terminate optimization if all gradient change less than e_rel
                 if e_rel is not None:
-                    crit = lambda x, x_: jnp.linalg.norm(x - x_) < e_rel * jnp.linalg.norm(x_)
-                    converged = tuple(
-                        crit(p, p_) for (p, p_) in zip(scene.parameters.values(), scene_.parameters.values()))
-                    if all(converged):
+                    crit = lambda x, x_: jnp.linalg.norm(x) < e_rel * jnp.linalg.norm(x_)
```
Comment (pmelchior) [inline scarlet2/scene.py:189]: You are assuming a specific set of `Parameter` instances to be given (spectrum and morphology in this case), but this may not be the case (or some of them may be frozen). I suggest to do this crit check more generally, namely by using `jax.tree_util.tree_reduce`, so that you can go through all of the pytree leaves automatically.

**Review by pmelchior — APPROVED** [review]
